src/editor/lib/commands/EntityRenameCommand.js

```diff
@@ -0,0 +1,61 @@
+import Events from '../Events.js';
+import { Command } from '../command.js';
+import { createUniqueId } from '../entity.js';
+
+export class EntityRenameCommand extends Command {
+  constructor(editor, entity) {
+    super(editor);
+
+    this.type = 'entityrename';
+    this.name = 'Rename Entity';
+    this.updatable = true; // Allow updating for consecutive renames
+
+    if (!entity.id) {
+      entity.id = createUniqueId();
+    }
+    this.entityId = entity.id;
+    this.oldName = entity.getAttribute('data-layer-name') || '';
+    this.newName = null; // Will be set during execute
```

Comment:
It needs to be set in the constructor otherwise you can't undo/redo

Reply:
undo/redo works in my testing?

Reply:
you're right it works, this.newName is set so it's fine afterwards